CRM/Contact/BAO/Query.php

```diff
@@ -5974,7 +5974,12 @@ public function optionValueQuery(
         $wc = "contact_a.$fieldName";
       }
       else {
-        $wc = "$tableName.id";
+        if ($field['name'] == 'on_hold') {
```

Comment:
Can you add a comment here to explain the special handling for future code editors..

Reply:
Done. I'm open to improvements in the wording.